spec/controllers/user_sessions_controller_spec.rb

```diff
@@ -12,7 +12,18 @@
 
   describe "#create" do
     let(:post_create) do
-      post :create, user_session: { email: user.email , password: user.password }
+      post :create, user_session: { email: user.email, password: user.password }
+    end
+
+    context 'when the user is approved' do
+      let(:user) { users(:affiliate_manager) }
```

Comment:
You don't need a separate context for this, because this should apply to all users. You can just move the  `let(:user)` line to below `describe "#create` so that the `user` variable will be available in all the `create` specs.:
```ruby
  describe "#create" do
    let(:user) { users(:affiliate_manager) }
    let(:post_create) do
      post :create, user_session: { email: user.email, password: user.password }
    end

    it 'filters passwords in the logfile' do
      allow(Rails.logger).to receive(:info)
      expect(Rails.logger).to receive(:info).
        with(/ \"password\"=>\"\[FILTERED\]\"/)
      post_create
    end
```